src/cargo/ops/cargo_update.rs

```diff
@@ -207,6 +214,251 @@ pub fn print_lockfile_changes(
     }
 }
 
+pub fn upgrade_manifests(
+    ws: &mut Workspace<'_>,
+    opts: &UpdateOptions<'_>,
+) -> CargoResult<HashMap<String, Version>> {
+    let mut upgrades = HashMap::new();
+    let mut upgrade_messages = HashSet::new();
+
+    if !opts.breaking {
+        return Ok(upgrades);
+    }
+
+    // Updates often require a lot of modifications to the registry, so ensure
+    // that we're synchronized against other Cargos.
+    let _lock = ws
+        .gctx()
+        .acquire_package_cache_lock(CacheLockMode::DownloadExclusive)?;
+
+    let mut registry = PackageRegistry::new(opts.gctx)?;
```

Comment:
The caller handles the policy of checking `opts.breaking`, maybe we should not do that here and stop accepting `UpdateOptions`, only accepting `GlobalContext`

Reply:
I can remove that extra check, but not `UpdateOptions`, as we are also using `opts.to_update`.

Reply:
Check removed.

Reply:
Honestly,. I would just pass the parameters directly and not use `opts` at all in this case (which means `UpdateOptions` doesn't need `breaking`).

The alternative is to move all of the logic in the `command/update.rs` into the `ops` and `UpdateOptions` can be passed to that but the lower level functions probably shouldn't accept it.

Reply:
OK, I implemented the first option (not passing `upgrade_opts`).